snap-csv-dataio/src/main/java/org/esa/snap/csv/dataio/writer/CsvProductWriter.java

```diff
@@ -132,61 +136,41 @@ public void writeBandRasterData(Band sourceBand, int sourceOffsetX, int sourceOf
         if (productWritten) {
             return;
         }
-        final Band[] bands = getSourceProduct().getBands();
-        final DataBuffer[] bandDataBuffers = new DataBuffer[bands.length];
-        final int[] bandTypes = new int[bands.length];
-        for (int i = 0; i < bands.length; i++) {
-            final Band band = bands[i];
-            bandDataBuffers[i] = band.getGeophysicalImage().getData().getDataBuffer();
-            bandTypes[i] = bandDataBuffers[i].getDataType();
-        }
-
+        final RasterDataNode[] bands = getSourceProduct().getBands();
         final TiePointGrid[] tiePointGrids = getSourceProduct().getTiePointGrids();
-        final DataBuffer[] tpgDataBuffers = new DataBuffer[tiePointGrids.length];
-        final int[] tpgTypes = new int[tiePointGrids.length];
-        for (int i = 0; i < tiePointGrids.length; i++) {
-            final TiePointGrid tpg = tiePointGrids[i];
-            tpgDataBuffers[i] = tpg.getGeophysicalImage().getData().getDataBuffer();
-            tpgTypes[i] = tpgDataBuffers[i].getDataType();
+        final RasterDataNode[] rasterDataNodes = Arrays.copyOf(bands, bands.length + tiePointGrids.length);
+        System.arraycopy(tiePointGrids, 0, rasterDataNodes, bands.length, tiePointGrids.length );
+
+        final int[] dataTypes = new int[bands.length];
+
+        final Raster[] dataRasters = new Raster[rasterDataNodes.length];
+        // todo: For big products this will not work because the hole scene is allocated. But no one wants to
+        //  export big scenes to ASCII. So this is minor and can be adressed when the need arises.
+        Rectangle dataRect = new Rectangle(getSourceProduct().getSceneRasterWidth(), getSourceProduct().getSceneRasterHeight());
+        for (int i = 0; i < rasterDataNodes.length; i++) {
+            dataRasters[i] = rasterDataNodes[i].getGeophysicalImage().getData(dataRect);
+            dataTypes[i] = dataRasters[i].getDataBuffer().getDataType();
         }
 
+
```

Comment:
Remove extra line. It looks ugly.

Reply:
In the first test class, the setup method is called for each test method. This is not necessary for the new test

Reply:
Okay, I see.